lib/Sema/TreeTransform.h

```diff
@@ -11535,6 +11539,12 @@ TreeTransform<Derived>::TransformRangeBoundsExpr(RangeBoundsExpr *E) {
                                              E->getRParenLoc());
 }
 
+template<typename Derived>
+ExprResult
+TreeTransform<Derived>::TransformInteropTypeBoundsAnnot(InteropTypeBoundsAnnot *E) {
```

Comment:
Don't we need to "recur" on the type inside the annotation?  I'm relatively unfamiliar with this code, but I'm guessing there's some facility that might be used, say, to rename or transform types within an AST.  I'd assume we need to avoid pruning that recursive walk at these annotations, in case the type inside it matches a type of interest for the caller.

Feel free to ignore if I'm making no sense here.